src/main.rs

```diff
@@ -1179,6 +1221,91 @@ mod tests {
         });
     }
 
+    #[cfg_attr(not(feature = "mmio"), test)]
+    fn test_vhost_user_blk() {
+        test_block!(tb, "", {
+            let mut clear = ClearDiskConfig::new();
+            let guest = Guest::new(&mut clear);
+
+            let (mut daemon_child, vubd_socket_path) = prepare_vubd(&guest.tmp_dir);
+
+            let mut cloud_child = Command::new("target/debug/cloud-hypervisor")
+                .args(&["--cpus", "4"])
+                .args(&["--memory", "size=512M,file=/dev/shm"])
+                .args(&["--kernel", guest.fw_path.as_str()])
+                .args(&[
+                    "--disk",
+                    guest
+                        .disk_config
+                        .disk(DiskType::OperatingSystem)
+                        .unwrap()
+                        .as_str(),
+                    guest
+                        .disk_config
+                        .disk(DiskType::CloudInit)
+                        .unwrap()
+                        .as_str(),
+                ])
+                .args(&["--net", guest.default_net_string().as_str()])
+                .args(&[
+                    "--vhost-user-blk",
+                    format!(
+                        "sock={},num_queues=1,queue_size=128,wce=true",
+                        vubd_socket_path
+                    )
+                    .as_str(),
+                ])
+                .spawn()
+                .unwrap();
+
+            thread::sleep(std::time::Duration::new(20, 0));
+
+            // Check both if /dev/vdc exists and if the block size is 16M.
+            aver_eq!(
+                tb,
+                guest
+                    .ssh_command("lsblk | grep vdc | grep -c 16M")
+                    .unwrap_or_default()
+                    .trim()
+                    .parse::<u32>()
+                    .unwrap_or_default(),
+                1
+            );
+
+            // Mount the device
+            guest.ssh_command("mkdir mount_image").unwrap();
```

Comment:
Mustn't call `.unwrap()` after an `aver_eq!`. This can lead to hard to debug SIGILL crashes (nested panics.)